src/Common/src/Interop/Unix/libssl/SslConnectionInfo.cs

```diff
@@ -190,5 +202,32 @@ private static HashAlgorithmType MapHashAlgorithmType(Interop.libssl.DataHashAlg
                     return HashAlgorithmType.None;
             }
         }
+        
+        private static int GetHashAlgorithmBitSize(Interop.libssl.DataHashAlgorithm mac)
+        {
+            switch (mac)
+            {
+                case Interop.libssl.DataHashAlgorithm.SSL_MD5:
+                    return MD5_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA1:
+                    return SHA1_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_GOST94:
+                    return GOST_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_GOST89MAC:
+                    return GOST_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA256:
+                    return SHA256_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA384:
+                    return SHA384_HashKeySize;
+
+                default:
+                    return 0;
```

Comment:
@shrutigarg Given that this has a not-understood behavior on OSX due to the struct being copied into C# (vs shimmed), I think it's best if you shim this up.

Modern libssl exposes what you want in  `int SSL_CIPHER_get_digest_nid(const SSL_CIPHER *c);` (didn't exist in 0.9.8).

You probably want something like
```C++
extern "C" int32_t GetSslCipherMacInfo(SSL* ssl, int32_t** macNid, int32_t** macSizeBytes)
{
    // Test ssl, algNid, algSizeBytes for validity.  This is just a sample.

    SSL_CIPHER* cipher = SSL_get_current_cipher(ssl);

    if (!cipher)
    {
        // do something
    }

    int cipherMac;
    
#if IMNOTSUREWHAT // OpenSSL 0.9.8
    switch (cipher->algorithms & SSL_MAC_MASK)
    {
        case SSL_MD5:
            cipherMac = NID_md5;
            break;
        case SSL_SHA1:
            cipherMac = NID_sha1;
            break;
        default:
            // Handle this unknown case
            break;
    }
#else
    cipherMac = SSL_CIPHER_get_digest_nid(cipher);
#endif

    if (cipherMac == NID_unknown)
    {
        assert(cipher->algorithm_mac == SSL_AEAD);
        *macSizeBytes = 0;
    }
    else
    {
        *macSizeBytes = EVP_MD_size(EVP_get_digestbynid(cipherMac));
    }

    *macNId = cipherMac;
    return 1;
}
```